From htejun@gmail.com  Tue May 29 04:50:15 2007
Cc: Tejun Heo <htejun@gmail.com>
Subject: [PATCH 5/5] sysfs: fix root sysfs_dirent -> root dentry association
Date: Tue, 29 May 2007 01:10:26 +0900
Message-Id: <11803686261278-git-send-email-htejun@gmail.com>
To: gregkh@suse.de, dmitry.torokhov@gmail.com,
	cornelia.huck@de.ibm.com, oneukum@suse.de, rpurdie@rpsys.net,
	stern@rowland.harvard.edu, maneesh@in.ibm.com,
	linux-kernel@vger.kernel.org, htejun@gmail.com
From: Tejun Heo <htejun@gmail.com>

The root sysfs_dirent didn't point to the root dentry fix it.

Signed-off-by: Tejun Heo <htejun@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/sysfs/mount.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/sysfs/mount.c
+++ b/fs/sysfs/mount.c
@@ -65,6 +65,7 @@ static int sysfs_fill_super(struct super
 		iput(inode);
 		return -ENOMEM;
 	}
+	sysfs_root.s_dentry = root;
 	root->d_fsdata = &sysfs_root;
 	sb->s_root = root;
 	return 0;
